modules/light-clients/06-solomachine/light_client_module_test.go

```diff
@@ -1,18 +1,797 @@
 package solomachine_test
 
 import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
 	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v8/modules/core/23-commitment/types"
 	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+	ibcerrors "github.com/cosmos/ibc-go/v8/modules/core/errors"
 	"github.com/cosmos/ibc-go/v8/modules/core/exported"
 	solomachine "github.com/cosmos/ibc-go/v8/modules/light-clients/06-solomachine"
 	ibctesting "github.com/cosmos/ibc-go/v8/testing"
+	ibcmock "github.com/cosmos/ibc-go/v8/testing/mock"
 )
 
 const (
 	smClientID   = "06-solomachine-100"
 	wasmClientID = "08-wasm-0"
 )
 
+func (suite *SoloMachineTestSuite) TestStatus() {
+	clientID := suite.chainA.App.GetIBCKeeper().ClientKeeper.GenerateClientIdentifier(suite.chainA.GetContext(), exported.Solomachine)
+	clientState := suite.solomachine.ClientState()
+
+	// Set a client state in store.
+	suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientState(suite.chainA.GetContext(), clientID, clientState)
+
+	lightClientModule, found := suite.chainA.App.GetIBCKeeper().ClientKeeper.Route(clientID)
+	suite.Require().True(found)
+
+	status := lightClientModule.Status(suite.chainA.GetContext(), clientID)
+
+	// solo machine discards arguments
+	suite.Require().Equal(exported.Active, status)
+
+	// freeze solo machine and update it in store.
+	clientState.IsFrozen = true
+	suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientState(suite.chainA.GetContext(), clientID, clientState)
+
+	status = clientState.Status(suite.chainA.GetContext(), nil, nil)
+	suite.Require().Equal(exported.Frozen, status)
+}
+
+func (suite *SoloMachineTestSuite) TestGetTimestampAtHeight() {
+	clientID := suite.chainA.App.GetIBCKeeper().ClientKeeper.GenerateClientIdentifier(suite.chainA.GetContext(), exported.Solomachine)
```

Comment:
Similar comment: instead of generating a new client ID and set a client state for it, we could use the client ID and client state that are created in `SetupTest`. I think we would need to change the client IDs that are generated are the moment, because [they don't have the right format](https://github.com/cosmos/ibc-go/blob/5514d56284135be44dd064a1ffbfe5ae5b607b4a/modules/light-clients/06-solomachine/solomachine_test.go#L48-L49), but if we do that, then we wouldn't need to set an extra client state, I think.